python/ray/data/datasource/partitioning.py

```diff
@@ -353,14 +359,17 @@ def _parse_hive_path(self, dir_path: str) -> Dict[str, str]:
         kv_pairs = [d.split("=") for d in dirs] if dirs else []
```

Comment:
We should open an issue for a follow-up PR around supporting basic value parsing beyond just string support. Easiest route would be requiring a `pa.Schema` to be provided to the partitioning specification for the partition columns, which would give the partition column types and allow us to use `pa.scalar(part_value).cast(schema.field(part_key).type)` to cast.

Reply:
Done https://github.com/ray-project/ray/issues/28720